packages/firebase_crashlytics/android/src/main/java/io/flutter/plugins/firebase/crashlytics/firebasecrashlytics/FirebaseCrashlyticsPlugin.java

```diff
@@ -0,0 +1,92 @@
+      List<String> lines = (List<String>) call.argument("stackTraceLines");
+      List<StackTraceElement> elements = new ArrayList<>();
+      for (String line : lines) {
+        elements.add(generateStackTraceElement(line));
+      }
+      exception.setStackTrace(elements.toArray(new StackTraceElement[elements.size()]));
+
+      Crashlytics.setString("exception", (String) call.argument("exception"));
+      Crashlytics.setString("stackTrace", (String) call.argument("stackTrace"));
+      Crashlytics.logException(exception);
+      result.success("Error reported to Crashlytics.");
+    } else if (call.method.equals("Crashlytics#isDebuggable")) {
+      result.success(Fabric.isDebuggable());
+    } else if (call.method.equals("Crashlytics#getVersion")) {
+      result.success(Crashlytics.getInstance().getVersion());
+    } else if (call.method.equals("Crashlytics#setInt")) {
+      Crashlytics.setInt((String) call.argument("key"), (int) call.argument("value"));
+      result.success(null);
+    } else if (call.method.equals("Crashlytics#setDouble")) {
+      Crashlytics.setDouble((String) call.argument("key"), (double) call.argument("value"));
+      result.success(null);
+    } else if (call.method.equals("Crashlytics#setString")) {
+      Crashlytics.setString((String) call.argument("key"), (String) call.argument("value"));
+      result.success(null);
+    } else if (call.method.equals("Crashlytics#setBool")) {
+      Crashlytics.setBool((String) call.argument("key"), (boolean) call.argument("value"));
+      result.success(null);
+    } else if (call.method.equals("Crashlytics#log")) {
+      Crashlytics.log((String) call.argument("msg"));
+      result.success(null);
+    } else if (call.method.equals("Crashlytics#setUserEmail")) {
+      Crashlytics.setUserEmail((String) call.argument("email"));
+      result.success(null);
+    } else if (call.method.equals("Crashlytics#setUserIdentifier")) {
+      Crashlytics.setUserIdentifier((String) call.argument("identifier"));
+      result.success(null);
+    } else if (call.method.equals("Crashlytics#setUserName")) {
+      Crashlytics.setUserName((String) call.argument("name"));
+      result.success(null);
+    } else {
+      result.notImplemented();
+    }
+  }
+
+  /**
+   * Extract StackTraceElement from line from Dart stack trace. Incoming line is expected in the
+   * following format:
+   *
+   * <p>FILE_PATH LINE_NUMBER:CHARACTER_NUMBER METHOD_NAME
+   *
+   * @param line Line from Dart stack trace.
+   * @return Stack trace element to be used as part of an Exception stack trace.
+   */
+  private StackTraceElement generateStackTraceElement(String line) {
+    // Split line on white spaces.
+    String[] lineParts = line.split("\\s+");
+    String fileName = lineParts[0].trim();
+    String lineNumber = lineParts[1].trim().substring(0, lineParts[1].indexOf(":"));
+    String className = lineParts[2].trim().substring(0, lineParts[2].indexOf("."));
```

Comment:
It's possible that if `trim()` removes spaces from the start and then the call to `indexOf` will report an incorrect position and these line's can throw `IndexOutOfBoundsException`.

Reply:
@slightfoot I think it could be an issue, I'll add some checks to make this safer.

Reply:
Called trim after the string operations.